mqtt/mqtt-broker/src/session.rs

```diff
@@ -350,7 +358,19 @@ impl SessionState {
                 let event = self.prepare_to_send(&publication)?;
                 Ok(Some(event))
             } else {
-                self.waiting_to_be_sent.push_back(publication);
+                if self.waiting_to_be_sent.len() > self.config.max_queued_messages() {
+                    match self.config.when_full() {
+                        QueueFullAction::DropNew => {
+                            return Ok(None);
+                        }
+                        QueueFullAction::DropOld => {
+                            drop(self.waiting_to_be_sent.pop_front());
```

Comment:
I think in general we should avoid calling `drop` manually
```suggestion
                            let _ = self.waiting_to_be_sent.pop_front();
```

Reply:
Sure, I thought it would make it more clear that we want to drop the message.